staging/src/k8s.io/pod-security-admission/policy/visitor.go

```diff
@@ -35,3 +43,30 @@ func visitContainers(podSpec *corev1.PodSpec, visitor ContainerVisitor) {
 		visitor((*corev1.Container)(&podSpec.EphemeralContainers[i].EphemeralContainerCommon))
 	}
 }
+
+// ContainerVisitorWithPath is called with each container and the field.Path to that container,
+// and returns true if visiting should continue.
+type ContainerVisitorWithPath func(container *corev1.Container, path *field.Path, errListHandler ErrListHandler) bool
+
+// visitContainersWithPath invokes the visitor function with a pointer to the spec
+// of every container in the given pod spec and the field.Path to that container.
+// If visitor returns false, visiting is short-circuited. VisitContainersWithPath returns true if visiting completes,
+// false if visiting was short-circuited.
+func visitContainersWithPath(podSpec *corev1.PodSpec, visitor ContainerVisitorWithPath, errListHandler ErrListHandler) bool {
```

Comment:
Why do we need to return a `bool` now ? And why are we returning early ?

Reply:
Do we need both `visitContainers` and `visitContainersWithPath` ?

Reply:
> Why do we need to return a `bool` now ? And why are we returning early ?

Yes, there should be no scenario where I need to return early now, I just kept the original structure of the `visitContainersWithPath` function before.

Reply:
> Do we need both `visitContainers` and `visitContainersWithPath` ?

Yes, `visitContainersWithPath` should cover all our needs